docarray/array/abstract_array.py

```diff
@@ -34,7 +35,10 @@ def __repr__(self):
         return f'<{self.__class__.__name__} (length={len(self)})>'
 
     @classmethod
-    def __class_getitem__(cls, item: Type[BaseDocument]):
+    def __class_getitem__(cls, item: Union[Type[BaseDocument], TypeVar, str]):
+        if not isinstance(item, type):
+            return Generic.__class_getitem__.__func__(cls, item)  # type: ignore
```

Comment:
How did you know this is what is required?

Reply:
> is this even doing anything?

this is not doing anything else than checking the the `item` is valid Generic. I think it does not harm to keep it

Reply:
> > is this even doing anything?
> 
> this is not doing anything else than checking the the `item` is valid Generic. I think it does not harm to keep it

Crashed in Python 3.12.0
```
    return Generic.__class_getitem__.__func__(cls, item)  # type: ignore
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'builtin_function_or_method' object has no attribute '__func__'. Did you mean: '__doc__'?
```